### PR TITLE
[W5.4][W15-A3] Hoon Chee Yu

### PR DESCRIPTION
**Pull request by hooncheeyu**

### CODE REVIEW

File: src/seedu/addressbook/commands/UpdateCommand.java [hunk 1]
```diff
@@ -0,0 +1,73 @@
...
+import seedu.addressbook.common.Messages;
+import seedu.addressbook.data.exception.IllegalValueException;
+import seedu.addressbook.data.person.*;
+import seedu.addressbook.data.tag.Tag;
+import seedu.addressbook.data.tag.UniqueTagList;
+
+import java.util.HashSet;
+import java.util.Set;
+
+public class UpdateCommand extends Command{
```
Comment (okkhoy) [inline src/seedu/addressbook/commands/UpdateCommand.java]: need space between the word and `{`

File: src/seedu/addressbook/commands/UpdateCommand.java [hunk 2]
```diff
@@ -0,0 +1,73 @@
+package seedu.addressbook.commands;
+
+import seedu.addressbook.common.Messages;
+import seedu.addressbook.data.exception.IllegalValueException;
+import seedu.addressbook.data.person.*;
```
Comment (okkhoy) [inline src/seedu/addressbook/commands/UpdateCommand.java]: Don't import using `*` import the classes explicitly, only what is necessary

File: src/seedu/addressbook/data/AddressBook.java [hunk 3]
```diff
@@ -84,6 +84,18 @@ public void addPerson(Person toAdd) throws DuplicatePersonException {
     }
 
     /**
+     * Updates a person from the address book.
```
Comment (okkhoy) [inline src/seedu/addressbook/data/AddressBook.java]: Updates a person *in* the address book?

File: src/seedu/addressbook/parser/Parser.java [hunk 4]
```diff
@@ -138,6 +131,86 @@ private Command prepareAdd(String args) {
...
+            );
+
+        } catch (ParseException pe) {
+            return new IncorrectCommand(String.format(MESSAGE_INVALID_COMMAND_FORMAT, DeleteCommand.MESSAGE_USAGE));
+        } catch (NumberFormatException nfe) {
+            return new IncorrectCommand(MESSAGE_INVALID_PERSON_DISPLAYED_INDEX);
+        }catch (IllegalValueException ive) {
+            return new IncorrectCommand(ive.getMessage());
+        }
+        /*try {
```
Comment (okkhoy) [inline src/seedu/addressbook/parser/Parser.java]: don't retain commented code; delete it. you can always recover the code from VCS

**Comment by okkhoy** [comment]
@hooncheeyu 
Some comments added. Please ack & close the PR after reading comments.

**Comment by okkhoy** [comment]
@hooncheeyu 
Since there are notable missing parts, this submission does not meet our requirements. We suggest you have another go as explained in the recent announcement.

**Comment by hooncheeyu** [comment]
@okkhoy 
Thanks for the review, I will follow the recent announcement and update addressbook level 3. Thanks

**Pull request closed without merging**